app/components/ClientUtilityFunctions.tsx

```diff
@@ -0,0 +1,23 @@
+'use client'
+import { useRouter } from 'next/navigation';
+import styles from "../../styles/Profile.module.css"
+/**
+ * Conditionally renders the button if the profile belongs to the current user
+ * session else renders a follow button
+ * 
+ * @param isUser - true if current session user matches page, false otherwise
+ */
+export function EditProfileAndFollowButton({ user, isUser }: { user: string, isUser: boolean }) {
+    const { push } = useRouter();
+
+    const handleClick = (e) => {
+        e.preventDefault();
+        push(`${user}/edit-profile`);
+    }
```

Comment:
Is it possible to use relative URLs with `router.push()`? If so, you wouldn't have to pass the `user` prop into the button.

```suggestion
export function EditProfileAndFollowButton(isUser: boolean) {
    const { push } = useRouter();

    const handleClick = (e) => {
        e.preventDefault();
        push('/edit-profile');
    }
```